File: app/src/main/java/net/numa08/gochisou/GochisouApplication.java
package net.numa08.gochisou;

import android.app.Application;
import android.support.annotation.VisibleForTesting;

import net.numa08.gochisou.presentation.internal.di.components.ApplicationComponent;
import net.numa08.gochisou.presentation.internal.di.components.DaggerApplicationComponent;
import net.numa08.gochisou.presentation.internal.di.modules.ApplicationModule;


public class GochisouApplication extends Application {

    private static GochisouApplication self;

    public static GochisouApplication application() {
        return self;
    }

    @VisibleForTesting
    public ApplicationComponent applicationComponent;

    public ApplicationComponent getApplicationComponent() {
        return applicationComponent;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        self = this;
        applicationComponent = DaggerApplicationComponent
                .builder()
                .applicationModule(new ApplicationModule(this))
                .build();
    }
}

File: app/src/main/kotlin/net/numa08/gochisou/GochisouApplication.kt
package net.numa08.gochisou

import android.app.Application
import android.support.annotation.VisibleForTesting

import net.numa08.gochisou.presentation.internal.di.components.ApplicationComponent
import net.numa08.gochisou.presentation.internal.di.components.DaggerApplicationComponent
import net.numa08.gochisou.presentation.internal.di.modules.ApplicationModule


class GochisouApplication : Application() {

    @VisibleForTesting
    var applicationComponent: ApplicationComponent? = null

    override fun onCreate() {
        super.onCreate()
        application = this
        applicationComponent = DaggerApplicationComponent.builder().applicationModule(ApplicationModule(this)).build()
    }

    companion object {

        var application: GochisouApplication? = null

    }
}
